packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -79,15 +79,13 @@ class FlutterDevice {
     vmServices = localVmServices;
   }
 
-  Future<void> refreshViews() {
+  Future<void> refreshViews() async {
     if (vmServices == null || vmServices.isEmpty)
-      return Future<void>.value(null);
+      return;
     final List<Future<void>> futures = <Future<void>>[];
     for (VMService service in vmServices)
-      futures.add(service.vm.refreshViews());
-    final Completer<void> completer = Completer<void>();
-    Future.wait(futures).whenComplete(() => completer.complete(null)); // ignore: unawaited_futures
-    return completer.future;
+      futures.add(service.vm.refreshViews(waitForViews: true));
+    await Future.wait(futures);
```

Comment:
I think my intent was not to `await `here, so refreshViews could continue without blocking. Was there a reason you put `await` back?

Reply:
The caller of `refreshViews()` can choose to not await the returned future.  The use of async/await here is semantically equivalent to what you were trying to do with the `Completer`.

Reply:
Ok, got it.

Reply:
yeah these are identical semantically. I happened to have already made this change on my branch; I didn't put the await back so much as have to resolve the merge. :-)